src/main.rs

```diff
@@ -89,87 +95,131 @@ fn main() -> Result<(), Error> {
-            )?,
-            &*consul_configs::create_contract_abi(
-                "NectarToken",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            &*consul_configs::wait_or_get(
-                "homechain",
-                "erc20_relay_address",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            &*consul_configs::create_contract_abi(
-                "ERC20Relay",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            settings.relay.homechain.free,
-            settings.relay.confirmations,
-        )?,
-        Network::sidechain(
-            side_ws.clone(),
-            &settings.relay.account,
-            &*consul_configs::wait_or_get(
-                "sidechain",
-                "nectar_token_address",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            &*consul_configs::create_contract_abi(
-                "NectarToken",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            &*consul_configs::wait_or_get(
-                "sidechain",
-                "erc20_relay_address",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            &*consul_configs::create_contract_abi(
-                "ERC20Relay",
-                &settings.relay.consul,
-                &settings.relay.consul_token,
-                &settings.relay.community,
-            )?,
-            settings.relay.sidechain.free,
-            settings.relay.confirmations,
-            settings.relay.anchor_frequency,
-        )?,
-    );
-
-    // Unlock accounts now
-    eloop.run(relay.unlock(&settings.relay.password))?;
+        .map_err(SyncFailure::new)
+        .unwrap();
 
     // Run the relay
-    handle.spawn(relay.run(&handle));
+    handle.spawn(run(handle.clone(), settings, home_ws, side_ws));
     while running.load(Ordering::SeqCst) {
         eloop.turn(Some(Duration::from_secs(1)));
     }
 
     Ok(())
 }
+
+fn run(handle: reactor::Handle, settings: Settings, home_ws: web3::transports::WebSocket, side_ws: web3::transports::WebSocket) -> impl Future<Item = (), Error = ()>  {
+
+    let account = utils::clean_0x(&settings.relay.account)
+        .parse()
+        .or_else(|_| Err(OperationError::InvalidAddress(settings.relay.account.clone())))
+        .unwrap();
+
+    let home_web3 = Web3::new(home_ws.clone());
+    let side_web3 = Web3::new(side_ws.clone());
+
+    home_web3.eth().transaction_count(account, None)
```

Comment:
This is fine for now but when we merge it let's also file an issue about nonce-resync, judging from polyswarm-client experience that's gonna be an issue